File: src/main/java/io/github/bot/terminal/application/domains/bot_commands/entity/Cmd.java
package io.github.bot.terminal.application.domains.bot_commands.entity;

import lombok.Data;
import org.apache.commons.lang3.StringUtils;

@Data
public class Cmd {

    public static String EMPTY = "_CMD_EMPTY_";

    private String cmd;

    public Cmd(String cmd) {
        this.cmd = StringUtils.defaultString(cmd, EMPTY).trim().toLowerCase();
    }

    @Override
    public String toString() {
        return this.cmd;
    }
}

File: src/main/kotlin/io/github/bot/terminal/application/domains/bot_commands/entity/Cmd.kt
package io.github.bot.terminal.application.domains.bot_commands.entity

import org.apache.commons.lang3.StringUtils

class Cmd(cmd: String) {
    private val cmd: String
    init {
        this.cmd = StringUtils.defaultString(cmd, EMPTY).trim { it <= ' ' }.toLowerCase()
    }

    override fun toString(): String {
        return cmd
    }

    fun isNotEmpty(): Boolean = cmd != EMPTY

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Cmd

        if (cmd != other.cmd) return false

        return true
    }

    override fun hashCode(): Int {
        return cmd.hashCode()
    }

    companion object {
        private var EMPTY = "_CMD_EMPTY_"
    }

}